### DISCUSSION TITLE
Narayana LRA support: How to pass userData

### DISCUSSION THREAD

**Discussion by ziomill** [quarkusio/quarkus#31374] [Q&A] [answered]
Hi all,
We would implement a SAGA by using the Narayana LRA support provided by Quarkus.

Anyway, by following the example provided at https://quarkus.io/guides/lra, we are not able to pass the "userData" through the @Compensate and @Complete callbacks.

Seeing the LRA official specs available [here](https://download.eclipse.org/microprofile/microprofile-lra-2.0-RC1/microprofile-lra-spec-2.0-RC1.html), it seems that passing custom data between callbacks is not possible. Is there a way to accomplish this use case?

Thank you

[a comment by an automated account is not shown]

**Comment by xstefank** [comment, marked as answer]
Hi @ziomill, in the current version of the spec, the passing of user data is not supported in terms of the defined protocol.

However, you can achieve this in this manner:

```
@Path("/lra")
public class MyLRAResource {

    private Map<URI, String> lraData = new HashMap<>();

    @GET
    @Path("/doInLRA")
    @LRA
    public Response doInLRA(@HeaderParam(LRA.LRA_HTTP_CONTEXT_HEADER) URI lraId) {
        lraData.put(lraId, "my data");
        return Response.ok().build();
    }

    @PUT
    @Compensate
    public Response compensateLRA(@HeaderParam(LRA.LRA_HTTP_CONTEXT_HEADER) URI lraId) {
        System.out.println(lraData.get(lraId));
        return Response.ok().build();
    }
}
```

Or use database, filesystem, or any other form of message. Hopefully this helps.
